editor/find_in_files.cpp

```diff
@@ -398,8 +398,25 @@ FindInFilesDialog::FindInFilesDialog() {
 }
 
 void FindInFilesDialog::set_search_text(const String &text) {
-	_search_text_line_edit->set_text(text);
-	_on_search_text_modified(text);
+	if (_mode == SEARCH_MODE) {
+		_on_search_text_modified(text);
+		if (!text.is_empty()) {
+			_search_text_line_edit->set_text(text);
+			_on_search_text_modified(text);
```

Comment:
You could call `_on_search_text_modified()` once, after the condtion.

Reply:
The first `_on_search_text_modified(text);` was unnecessary, I removed it.